mcs/class/System/Test/System.Net.Sockets/SocketTest.cs

```diff
@@ -4707,9 +4707,10 @@ public void ConnectAsyncUnhandledEx ()
 			socketArgs.RemoteEndPoint = endPoint;
 			socketArgs.Completed += (sender, e) => mre.Set ();
 
-			socket.ConnectAsync (socketArgs);
+			if (socket.ConnectAsync (socketArgs))
+				Assert.IsTrue (mre.WaitOne (1000), "ConnectedAsync timeout");
 
-			Assert.IsTrue (mre.WaitOne (1000), "ConnectedAsync timeout");
+			Assert.AreNotEqual (SocketError.Success, socketArgs.SocketError);
```

Comment:
just for my edification, these test changes are because the ConnectAsync() can now complete either synchronously or asynchronously depending on timing?